app/http.php

```diff
@@ -0,0 +1,366 @@
+    }
+
+    Console::error('[Error] Type: ' . get_class($error));
+    Console::error('[Error] Message: ' . $error->getMessage());
+    Console::error('[Error] File: ' . $error->getFile());
+    Console::error('[Error] Line: ' . $error->getLine());
+}
+
+Console::success('Waiting for executors to start...');
+
+$startupDelay = \intval(App::getEnv('OPEN_RUNTIMES_PROXY_STARTUP_DELAY', '0'));
+if ($startupDelay > 0) {
+    \sleep((int) ($startupDelay / 1000));
+}
+
+\go(function () use ($register) {
+    if (App::getEnv('OPEN_RUNTIMES_PROXY_HEALTHCHECK', 'enabled') === 'enabled') {
+        fetchExecutorsState(true);
+    } else {
+        $executorStates = $register->get('executorStates');
+
+        // If no health check, mark all as online
+        $executors = \explode(',', (string) App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTORS', ''));
+
+        foreach ($executors as $executor) {
+            $executorStates->set($executor, [
+                'status' => 'online',
+                'hostname' => $executor,
+                'state' =>  \json_encode([])
+            ]);
+        }
+    }
+});
+
+Console::log('State of executors at startup:');
+
+go(function () use ($register) {
+    $executorStates = $register->get('executorStates');
+
+    $executors = \explode(',', (string) App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTORS', ''));
+
+    foreach ($executors as $executor) {
+        $state = $executorStates->exists($executor) ? $executorStates->get($executor) : null;
+
+        if ($state === null) {
+            Console::warning('Executor ' . $executor . ' has unknown state.');
+        } else {
+            Console::log('Executor ' . $executor . ' is ' . ($state['status'] ?? 'unknown') . '.');
+        }
+    }
+});
+
+Swoole\Event::wait();
+
+// Wildcard action
+App::error()
+    ->inject('request')
+    ->inject('response')
+    ->inject('roundRobinAtomic')
+    ->inject('executorStates')
+    ->action(function (Request $request, Response $response, Atomic $roundRobinAtomic, Table $executorStates) {
+        $secretKey = \explode(' ', $request->getHeader('authorization', ''))[1] ?? '';
+
+        if (empty($secretKey)) {
+            throw new Exception('Incorrect proxy key.', 401);
+        }
+        if ($secretKey !== App::getEnv('OPEN_RUNTIMES_PROXY_SECRET', '')) {
+            throw new Exception('Incorrect proxy key.', 401);
+        }
+
+        $roundRobinIndex = $roundRobinAtomic->get();
+
+        $algoType = App::getEnv('OPEN_RUNTIMES_PROXY_ALGORITHM', '');
+        $algo = match ($algoType) {
+            'round-robin' => new RoundRobin($roundRobinIndex - 1), // Atomic indexes from 1. Balancing library indexes from 0. That's why -1
+            'random' => new Random(),
+            default => new Random()
+        };
+
+        $balancing = new Balancing($algo);
+
+        $balancing->addFilter(fn ($option) => $option->getState('status', 'offline') === 'online');
```

Comment:
Lines 209-218 should be using App::setResource. This will also help us decouple and abstract some of the logic to main the controller simpler to read, understand and maintain.

Reply:
Created `balancing` resource that takes care of this.